02_Util/src/networkconnection/authenticator/Authenticator.ts

```diff
@@ -36,40 +37,50 @@ export class Authenticator implements IAuthenticator {
   async authenticate(
     allowUnknownChargingStations: boolean,
     identifier: string,
-    username?: string,
-    password?: string,
+    username: string,
+    password: string,
   ): Promise<boolean> {
-    if (
-      !allowUnknownChargingStations &&
-      (await this._locationRepository.readChargingStationByStationId(
-        identifier,
-      )) === null
-    ) {
+    if (!allowUnknownChargingStations && !(await this._isKnown(identifier))) {
       this._logger.warn('Unknown identifier', identifier);
       return false;
-    } else if (await this._cache.get(identifier, CacheNamespace.Connections)) {
+    }
+
+    if (await this._isAlreadyConnected(identifier)) {
       this._logger.warn(
         'New connection attempted for already connected identifier',
         identifier,
       );
       return false;
-    } else if (username && password) {
```

Comment:
for extra security, we could check the security profile here instead of just assuming (i.e., SP1 and SP2 *must* have these values)? I think that may happen already in the Network connection class though